odd_collector/adapters/druid/adapter.py

```diff
@@ -0,0 +1,79 @@
+from typing import List, Type
+
+from odd_collector_sdk.domain.adapter import AbstractAdapter
+from odd_models.models import DataEntityList, DataEntity, DataEntityType, DataSet, DataSetField, DataSetFieldType
+
+from odd_collector.adapters.druid.mappers.client import DruidBaseClient, DruidClient
+from odd_collector.adapters.druid.mappers.column import Column
+from odd_collector.adapters.druid.mappers.generator import DruidGenerator
+from odd_collector.adapters.druid.mappers.table import Table
+from odd_collector.adapters.druid.mappers.types import TYPES_DRUID_TO_ODD
+from odd_collector.domain.plugin import DruidPlugin
+
+
+class Adapter(AbstractAdapter):
+    def __init__(self, config: DruidPlugin, client: Type[DruidBaseClient] = None) -> None:
+        self.__host = config.host
+        self.__port = config.port
+        self.__oddrn_generator = DruidGenerator(
+            host_settings=f"{self.__host}"
+        )
+        client = client or DruidClient
+        self.client = client(config)
+
+    def get_data_source_oddrn(self) -> str:
+        return self.__oddrn_generator.get_data_source_oddrn()
+
+    def get_data_entity_list(self) -> DataEntityList:
+        return DataEntityList(
+            data_source_oddrn=self.get_data_source_oddrn(),
+            items=(self.get_data_entities()),
+        )
+
+    def get_data_entities(self) -> List[DataEntity]:
+        # Fetch
+        tables: List[Table] = self.client.get_tables()
+        columns: List[Column] = self.client.get_columns()
+
+        # Set oddrn
+        [self.set_table_oddrns(table) for table in tables]
```

Comment:
It shouldn't work as you expect.
At the end of iteration **generator** will have state for the latest table.
Set and get new path for each data entity before mapping.

```python
def map_table(generator, table, ...) -> DataEntity:
    generator.set_oddrn_paths(**{"schemas": table.schema, "tables": table.name})
    return DataEntity(
       ...
       oddrn=generator.get_oddrn_by_path("tables")
       ...
)
```